### PR TITLE
Increase HTTP request timeout used by OIDC authenticator

### PR DESCRIPTION
**Pull request by gusevda**
## Hey, I just made a Pull Request!

Closes https://github.com/backstage/backstage/issues/23267.

This PR increases HTTP request timeout of 3500ms used by OIDC authenticator to 10000ms.

The new value is set on a per-request bases according to the [openid-client documentation](https://github.com/panva/node-openid-client/blob/main/docs/README.md#customizing-individual-http-requests). This requires to set custom HTTP options in four places for different requests but it saves us from overriding HTTP options globally which can affect other libraries.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/auth-backend-module-oidc-provider/src/authenticator.ts [hunk 1]
```diff
@@ -30,6 +31,10 @@ import {
   PassportOAuthPrivateInfo,
 } from '@backstage/plugin-auth-node';
 
+custom.setHttpOptionsDefaults({
+  timeout: 10000,
+});
```
Comment (Rugvip) [inline plugins/auth-backend-module-oidc-provider/src/authenticator.ts:36]: Could we use the approach where the `custom.http_options` symbol is set on the client instead? It's kinda hard to know what other usages there might be of `openid-client` already or in the long run, so setting this globally is a bit risky.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 just a nit on the approach

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Just a minor potential opportunity for cleanup

**Review by vinzscam — APPROVED** [review]

File: plugins/auth-backend-module-oidc-provider/src/authenticator.ts [hunk 2]
```diff
@@ -66,7 +76,11 @@ export const oidcAuthenticator = createOAuthAuthenticator({
     const initializedScope = config.getOptionalString('scope');
     const initializedPrompt = config.getOptionalString('prompt');
 
+    Issuer[custom.http_options] = httpOptionsProvider;
     const promise = Issuer.discover(metadataUrl).then(issuer => {
+      issuer[custom.http_options] = httpOptionsProvider;
+      issuer.Client[custom.http_options] = httpOptionsProvider;
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/auth-backend-module-oidc-provider/src/authenticator.ts:82]: Alright! Impact of having unnecessary ones here is pretty low anyway, let's :shipit:

**Review by Rugvip — APPROVED** [review]

**Comment by Rugvip** [comment]
Commits have DCO, so gonna merge even though the check is stuck

**Pull request merged**